src/components/FlowRunsDeleteButton.vue

```diff
@@ -29,26 +31,49 @@
   const { showModal, open, close } = useShowModal()
 
   const api = useWorkspaceApi()
+  const loading = ref(false)
 
   const deleteFlowRuns = async (flowRuns: string[]): Promise<void> => {
+    loading.value = true
+
     const toastMessage = computed(() => {
       if (flowRuns.length === 1) {
         return 'Flow run deleted'
       }
       return `${flowRuns.length} flow runs deleted`
     })
 
+    const promises = flowRuns.map(flowRunId => deleteFlowRun(flowRunId))
+    const values = await Promise.allSettled(promises)
+    const errors = values.filter(value => value.status === 'rejected').map(value => value.reason)
+
+    if (errors.length > 0) {
+      if (errors.length === 1) {
+        const message = getApiErrorMessage(errors[0], localization.error.delete('Flow Run'))
+        showToast(message, 'error')
+      } else {
+        showToast(`${errors.length} flow runs failed to delete`, 'error')
```

Comment:
```suggestion
        showToast(`${errors.length} flow run(s) failed to delete`, 'error')
```
technically this could be removing just a single run

Reply:
Correct, but if there's only a single error we'll toast the specific error message for that run. This only happens if there were multiple failures.